catalog/resource_sql_table.go

```diff
@@ -431,6 +467,24 @@ func columnChangesCustomizeDiff(d *schema.ResourceDiff) error {
 					return fmt.Errorf("changing the 'type' of an existing column is not supported")
 				}
 			}
+		} else {
+			oldColsNameToMap := make(map[string]map[string]interface{})
+			newColsNameToMap := make(map[string]map[string]interface{})
+			for _, oldCol := range oldCols {
+				oldColMap := oldCol.(map[string]interface{})
+				oldColsNameToMap[oldColMap["name"].(string)] = oldColMap
+			}
+			for _, newCol := range newCols {
+				newColMap := newCol.(map[string]interface{})
+				newColsNameToMap[newColMap["name"].(string)] = newColMap
+			}
+			for name, oldColMap := range oldColsNameToMap {
+				if newColMap, exists := newColsNameToMap[name]; exists {
+					if oldColMap["type"] != newColMap["type"] || oldColMap["nullable"] != newColMap["nullable"] || oldColMap["comment"] != newColMap["comment"] {
```

Comment:
This seems fragile.

Reply:
Yeah, I agree, next step could be to make it possible to directly access data through the go struct instead of the map, but I'll probably deal with this in the future PRs